WestLaw UK.js

```diff
@@ -0,0 +1,163 @@
+	"target": "^https://uk\\.westlaw\\.com/Document/.*",
+	"minVersion": "3.0",
+	"maxVersion": "",
+	"priority": 100,
+	"inRepository": true,
+	"translatorType": 4,
+	"browserSupport": "gcsibv",
+	"lastUpdated": "2021-01-06 11:36:52"
+}
+
+/*
+	***** BEGIN LICENSE BLOCK *****
+
+	Copyright © 2020 YOUR_NAME <- TODO
+	
+	This file is part of Zotero.
+
+	Zotero is free software: you can redistribute it and/or modify
+	it under the terms of the GNU Affero General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	Zotero is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+	GNU Affero General Public License for more details.
+
+	You should have received a copy of the GNU Affero General Public License
+	along with Zotero. If not, see <http://www.gnu.org/licenses/>.
+
+	***** END LICENSE BLOCK *****
+*/
+
+
+function detectWeb(doc) {
+	// TODO: adjust the logic here
+	if (doc.getElementsByClassName("kh_toc-list ukResearch_toc-list")[0].innerText.includes("Case")) {
+		return "case";
+	}
+	else if (doc.getElementsByClassName("kh_toc-list ukResearch_toc-list")[0].innerText.includes("Act")) {
+		return "statute";
+	}
+	else if (doc.title.includes("s.")) {
+		return "statuteSection";
+	}
+	else {
+		return false;
+	}
+}
+
+function doWeb(doc, url) {
+	if (detectWeb(doc, url) == "case") {
+		return scrapeCase(doc, url);
+	}
+	else if (detectWeb(doc, url) == "statute") {
+		return scrapeStatute(doc, url);
+	}
+	else if (detectWeb(doc, url) == "statuteSection") {
+		return scrapeStatuteSection(doc, url);
+	}
+	else {
+		return false;
+	}
+}
+
+
+function scrapeCase(doc, url) {
+	var translator = Zotero.loadTranslator('web');
+	// Embedded Metadata
+	translator.setTranslator('951c027d-74ac-47d4-a107-9c3069ab7b48');
+	// translator.setDocument(doc);
+	
+	translator.setHandler('itemDone', function (obj, item) {
+		// TODO adjust if needed:
+		
+		var citationArray = parseCitationList(doc.getElementsByClassName("co_docContentMetaField")[3].innerText);
+		
+		item.reporter = citationArray[3];
+		item.reporterVolume = citationArray[2];
+		item.firstPage = citationArray[4];
+		item.dateDecided = citationArray[1];
+		item.court = /Court\n(.+)/.exec(doc.getElementById("co_docContentCourt").innerText)[1];
+		item.abstractNote = "";
+		item.complete();
+		
+		
+	});
+
+	translator.getTranslatorObject(function (trans) {
+		trans.itemType = "case";
+		// TODO map additional meta tags here, or delete completely
+
+		trans.doWeb(doc, url);
+	});
+}
+
+function scrapeStatute(doc, url) {
+	var translator = Zotero.loadTranslator('web');
+	// Embedded Metadata
+	translator.setTranslator('951c027d-74ac-47d4-a107-9c3069ab7b48');
+	// translator.setDocument(doc);
+	
+	translator.setHandler('itemDone', function (obj, item) {
+		// TODO adjust if needed:
```

Comment:
as above, you can remove these placeholder comments

Reply:
All gone